cmd/describe/externalauthprovider/cmd.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalauthprovider
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/externalauthprovider"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "external-auth-provider",
+	Aliases: []string{"externalauthproviders", "externalauthprovider", "external-auth-providers"},
+	Short:   "Show details of an external authentication provider on a cluster",
+	Long:    "Show details of an external authentication provider on a cluster.",
+	Example: `  # Show details of an external authentication provider named "exauth" on a cluster named "mycluster"
+  rosa describe external-auth-provider exauth --cluster=mycluster `,
+	Run:    run,
+	Hidden: true,
+	Args:   cobra.MaximumNArgs(1),
+}
+
+var args struct {
+	name string
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.SortFlags = false
+	ocm.AddClusterFlag(Cmd)
+	output.AddFlag(Cmd)
+	flags.StringVar(
+		&args.name,
+		"name",
+		"",
+		"Name for the external authentication provider of the cluster to target",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithAWS().WithOCM()
+	defer r.Cleanup()
+	err := runWithRuntime(r, cmd, argv)
+	if err != nil {
+		r.Reporter.Errorf(err.Error())
+		os.Exit(1)
+	}
+}
+
+func runWithRuntime(r *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+	externalAuthId := args.name
+	// Allow the use also directly set the external authentication id as positional parameter
+	if len(argv) == 1 && !cmd.Flag("name").Changed {
+		externalAuthId = argv[0]
+	}
+	if externalAuthId == "" {
+		return fmt.Errorf("you need to specify an external authentication provider name with '--name' parameter")
+	}
+	clusterKey := r.GetClusterKey()
+	cluster := r.FetchCluster()
+	if cluster.State() != cmv1.ClusterStateReady {
+		return r.Reporter.Errorf("Cluster '%s' is not yet ready", clusterKey)
+	}
+
+	err := externalauthprovider.ValidateHCPCluster(cluster)
+	if err != nil {
+		r.Reporter.Errorf("%v", err)
+		os.Exit(1)
```

Comment:
@chenz4027 You don't need this `os.Exit`, you can just return the `error` from the line above 👍

Reply:
Thanks @robpblake , will refactor

Reply:
Updated the error